### PR TITLE
 sdk: Create new Node Service and Clean up docs

### LINKED ISSUE #527: sdk: Cluster calls need to be adjusted
* cluster.Enumerate() -> Returns id of clusters paired to the current one. Currently just one
* cluster.Inspect(id) -> Returns cluster info and its node ids cluster.Enumerate().
    * Id may be a parameter that we add when pairing is added
* node.Inspect(id) -> Return node information, which is what cluster.Inspect() today.
* node.Enumerate() -> Return all the ids of the nodes attached to the current cluster.

### PR DESCRIPTION
**Pull request by lpabon**
**What this PR does / why we need it**:

* Major updates to documentation
* Updated gRPC REST Gateway entry points
* Cleaned up Cluster APIs by creating new Node Service
* Added InspectCurrent() to both Cluster and Node services
* Changed name of OpenStorageVolume.Set() to Update() to match others
* Cleaned up errors to return NotFound when the key is not found.
* Removed VolumeSpec from Clone() since it was not needed.

**Which issue(s) this PR fixes** (optional)
Closes #527

### CODE REVIEW

File: api/api.proto [hunk 1]
```diff
@@ -805,7 +820,7 @@ service OpenStorageVolume {
   rpc Create(SdkVolumeCreateRequest)
     returns (SdkVolumeCreateResponse) {
       option(google.api.http) = {
-        post: "/v1/volume"
+        post: "/v1/volumes"
```
Comment (ram-infrac) [inline api/api.proto]: why **volumes** -> volume create/delete etc will always be for single operation right?

File: api/api.proto [hunk 2]
```diff
@@ -849,17 +864,18 @@ service OpenStorageVolume {
   rpc Enumerate(SdkVolumeEnumerateRequest)
     returns (SdkVolumeEnumerateResponse) {
       option(google.api.http) = {
-        post: "/v1/volume/enumerate"
+        post: "/v1/volumes/enumerate"
```
Comment (ram-infrac) [inline api/api.proto]: should be GET call

File: api/api.proto [hunk 3]
```diff
@@ -1412,82 +1476,94 @@ message SdkVolumeUnmountRequest {
...
   // Unique name of the volume. This will be used for idempotency.
   string name = 1;
-  // Parent volume id, if specified will create a new volume as a clone of the parent.
+  // Parent volume id or snapshot id will create a new volume as a clone of the parent.
   string parent_id = 2;
-  // Volume specification
-  VolumeSpec spec = 3;
 }
 
+// Defines the resonpse when creating a clone from a volume or a snapshot
```
Comment (ram-infrac) [inline api/api.proto]: nit : response

File: api/api.proto [hunk 4]
```diff
@@ -1796,70 +1950,91 @@ message SdkCloudBackupHistoryItem {
...
 
+// Empty response
 message SdkCloudBackupSchedDeleteResponse{
 }
 
+// Empty request
 message SdkCloudBackupSchedEnumerateRequest{
 }
 
+// Defies a response containing a map listing the schedules for volume
```
Comment (ram-infrac) [inline api/api.proto]: nit: Defines

File: api/api.proto (hunk 1 above)
Comment (lpabon) [reply api/api.proto]: It was decided in the review to make them all plural

File: api/api.proto (hunk 2 above)
Comment (lpabon) [reply api/api.proto]: I tried to make this one a get, but it sends data to the server (the labels in the locator), so it will not allow a GET

Comment (ram-infrac) [reply api/api.proto]: ahh, I know this issue

**Pull request merged**